graphics/font.h

```diff
@@ -269,6 +269,41 @@ class Font {
 	int wordWrapText(const Common::String &str, int maxWidth, Common::Array<Common::String> &lines, int initWidth = 0, uint32 mode = kWordWrapOnExplicitNewLines) const;
 	/** @overload */
 	int wordWrapText(const Common::U32String &str, int maxWidth, Common::Array<Common::U32String> &lines, int initWidth = 0, uint32 mode = kWordWrapOnExplicitNewLines) const;
+
+	/**
+	 * Initializes the scalesurface for scaling and the grayScaleMap for the given @p srcSize and @p newSize.
+	 *
+	 * @param srcSize
+	 * @param newSize.
+	 */
+	void initializeScaling(int srcSize, int newSize);
+
+	/**
+	 * Scales the single gylph at @p chr the given the @p scale and @p resGray and returns the scaled surface with updated @p resGray
+	 *
+	 * @param width     The width of the bouding box.
+	 * @param height    The height of the bounding box.
+	 * @param grayLevel The gray level.
+	 * @param chr       The character to scale.
+	 * @param xOffset   The x offset.
+	 * @param srcheight The height of the source.
+	 * @param srcwidth  The width of the source.
+	 * @param scale     The scale.
+	 * @param resGray   The resulting grayscale
+	 * 
+	 * @return scaledsurface.
+	 */
+	Surface *scaleSingleGlyph(int width, int height, int grayLevel, int chr, int xOffset, int srcheight, int srcwidth, float scale, int *resGray);
+
+	/**
+	 * Deinitializes the scalesurface and the grayScaleMap after scaling is done
+	 */
+	void deinitializeScaling();
```

Comment:
We discussed that instead of having it is a separate method, you'd better merge it with the destructor